unstructured/partition/text.py

```diff
@@ -26,52 +27,125 @@
     segments = []
-    for i in range(0, len(content), n):
-        segment = content[i : i + n]  # noqa: E203
-        segments.append(segment)
+    if len(content) < n * 2:
+        segments = list(_split_in_half_at_breakpoint(content))
+    else:
+        segments = textwrap.wrap(content, width=n)
     return segments
 
 
-def _split_to_fit_max_content(content: str, max_partition: int = 1500) -> List[str]:
+def split_content_to_fit_min_max(
+    content: str,
+    max_partition: Optional[int] = 1500,
+    min_partition: Optional[int] = 0,
+) -> List[str]:
     """Splits a section of content so that all of the elements fit into the
-    max partition window."""
+    max/min partition window."""
     sentences = sent_tokenize(content)
-    num_sentences = len(sentences)
-
+    # import pdb; pdb.set_trace()
     chunks = []
-    chunk = ""
-
-    for i, sentence in enumerate(sentences):
-        if len(sentence) > max_partition:
-            chunks.extend(_split_content_size_n(sentence, n=max_partition))
-
-        if len(chunk + " " + sentence) > max_partition:
-            chunks.append(chunk)
-            chunk = sentence
+    tmp = ""
+    for sentence in sentences:
+        if max_partition is not None and len(sentence) > max_partition:
+            if tmp:
+                chunks.append(tmp)
+                tmp = ""
+            segments = _split_content_size_n(sentence, n=max_partition)
+            chunks.extend(segments[:-1])
+            tmp = segments[-1]
+        elif (
+            min_partition is not None
+            and max_partition is not None
+            and len(sentence) >= min_partition
+        ):
+            if len(tmp + " " + sentence) > max_partition:
+                if len(tmp) >= min_partition:
+                    chunks.append(tmp)
+                    tmp = ""
+                elif not tmp:
+                    chunks.append(sentence)
+                else:
+                    chunks.extend(_split_content_size_n(tmp + " " + sentence, n=max_partition))
+            else:
+                if not tmp:
+                    chunks.append(sentence)
+                else:
+                    chunks.extend([tmp, sentence])
+                    tmp = ""
         else:
-            chunk += " " + sentence
-            if i == num_sentences - 1:
-                chunks.append(chunk)
+            tmp += " " + sentence
+            tmp = tmp.strip()
+
+    if tmp:
+        chunks.append(tmp)
 
     return chunks
 
 
+def combine_paragraphs_less_than_min(
+    split_paragraphs: List[str],
+    max_partition: Optional[int] = 1500,
+    min_partition: Optional[int] = 0,
+) -> List[str]:
+    """Combine paragraphs less than min_partition while not exceeding max_partition."""
+    file_content: List[str] = []
+    tmp_paragraph = ""
+    y = 0
+    for x, paragraph in enumerate(split_paragraphs):
+        if y > x:
+            continue
+        if min_partition is not None and len(paragraph) < min_partition:
```

Comment:
Same for the conditionals in this function, I think they would benefit from comments